demos/smart_classroom_demo/cpp/src/cnn.cpp

```diff
@@ -1,127 +1,127 @@
-    input_blob_name_ = in.begin()->first;
-
-    InferenceEngine::OutputsDataMap out = cnnNetwork.getOutputsInfo();
-    for (auto&& item : out) {
-        item.second->setPrecision(InferenceEngine::Precision::FP32);
-        output_blobs_names_.push_back(item.first);
-    }
 
-    try {
-        executable_network_ = config_.ie.LoadNetwork(cnnNetwork, config_.deviceName);
-    } catch (const InferenceEngine::Exception&) {  // in case the model does not work with dynamic batch
-        cnnNetwork.setBatchSize(1);
-        executable_network_ = config_.ie.LoadNetwork(cnnNetwork, config_.deviceName,
-            {{InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED, InferenceEngine::PluginConfigParams::NO}});
+    m_input_tensor_name = model->input().get_any_name();
+
+    ov::OutputVector outputs = model->outputs();
+    for (auto& item : outputs) {
+        const std::string name = item.get_any_name();
+        m_output_tensors_names.push_back(name);
     }
-    logExecNetworkInfo(executable_network_, config_.path_to_model, config_.deviceName, config_.model_type);
-    infer_request_ = executable_network_.CreateInferRequest();
+
+    ov::preprocess::PrePostProcessor ppp(model);
+
+    ppp.input().tensor()
+        .set_element_type(ov::element::f32)
+        .set_layout(m_modelLayout);
+
+    model = ppp.build();
+
+    slog::info << "PrePostProcessor configuration:" << slog::endl;
+    slog::info << ppp << slog::endl;
+
+    ov::set_batch(model, m_config.m_max_batch_size);
+
+    m_compiled_model = m_config.m_core.compile_model(model, m_config.m_deviceName);
+    logCompiledModelInfo(m_compiled_model, m_config.m_path_to_model, m_config.m_deviceName, m_config.m_model_type);
+
+    m_infer_request = m_compiled_model.create_infer_request();
 }
 
 void CnnDLSDKBase::InferBatch(
         const std::vector<cv::Mat>& frames,
-        const std::function<void(const InferenceEngine::BlobMap&, size_t)>& fetch_results) const {
-    InferenceEngine::Blob::Ptr input = infer_request_.GetBlob(input_blob_name_);
-    const size_t batch_size = input->getTensorDesc().getDims()[0];
+        const std::function<void(const std::map<std::string, ov::Tensor>&, size_t)>& fetch_results) const {
+    ov::Tensor input_tensor = m_infer_request.get_tensor(m_input_tensor_name);
 
     size_t num_imgs = frames.size();
-    for (size_t batch_i = 0; batch_i < num_imgs; batch_i += batch_size) {
-        const size_t current_batch_size = std::min(batch_size, num_imgs - batch_i);
-        for (size_t b = 0; b < current_batch_size; b++) {
-            matToBlob(frames[batch_i + b], input, b);
-        }
 
-        if (config_.max_batch_size != 1)
-            infer_request_.SetBatch(current_batch_size);
-        infer_request_.Infer();
+    // shrink tensor from default m_config.m_max_batch_size to actual num of images;
+    ov::Shape shape = input_tensor.get_shape();
+    shape[ov::layout::batch_idx(m_modelLayout)] = num_imgs;
```

Comment:
You didn't change actual tensor shape by modifying copy returned by `get_shape()`. You should shrink tensor size this way https://github.com/openvinotoolkit/open_model_zoo/blob/7bad22800d144422e9e6f0e7279ca1df532d10dc/demos/interactive_face_detection_demo/cpp/detectors.cpp#L199

Reply:
yes, that's right. fixed